src/App.jsx

```diff
@@ -14,33 +16,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [sortGoods, setSortGoods] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortedGoods = (field, reverse = false) => {
+    const sortedGood = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    switch (field) {
+      case 'alphabet':
+        sortedGood.sort((item1, item2) => item1.localeCompare(item2));
+        break;
+      case 'length':
+        sortedGood.sort((item1, item2) => item1.length - item2.length);
+        break;
+      default:
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (reverse) {
+      sortedGood.reverse();
+    }
+
+    return sortedGood;
+  };
+
+  const visibleGoods = sortedGoods(sortGoods, isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortGoods !== 'alphabet',
+          })}
+          onClick={() => setSortGoods('alphabet')}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortGoods !== 'length',
+          })}
+          onClick={() => setSortGoods('length')}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(prevValue => !prevValue)}
+        >
+          Reverse
+        </button>
+        {(sortGoods || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setIsReversed(false);
+              setSortGoods('');
+            }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here